### ISSUE TITLE
sending button is not working for send msg

### ISSUE THREAD

**Issue by karthiksibi5161** [anirudhbagri/whatsapp-bulk-messenger#20]
Failed to send message

**Comment by akhilrajs** [comment]
yes i have the same problem i tried switching the class name and also tried xpath but the problem remains . has anyone found any solution to this

**Comment by karthiksibi5161** [comment]
I can't resolve that error so my friends build own bulk messenger.

**Comment by akhilrajs** [comment]
Is it on GitHub bro

**Comment by karthiksibi5161** [comment]
No, it doesn't.

**Comment by mrityunjayshukla411** [comment]
Install Pyautogui using And
Change the automator.py code to :- 

`from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from time import sleep
from urllib.parse import quote
import os
import pyautogui

options = Options()
options.add_experimental_option("excludeSwitches", ["enable-logging"])
options.add_argument("--profile-directory=Default")
# options.add_argument("--user-data-dir=/var/tmp/chrome_user_data")

os.system("")
os.environ["WDM_LOG_LEVEL"] = "0"


class style():
    BLACK = '\033[30m'
    RED = '\033[31m'
    GREEN = '\033[32m'
    YELLOW = '\033[33m'
    BLUE = '\033[34m'
    MAGENTA = '\033[35m'
    CYAN = '\033[36m'
    WHITE = '\033[37m'
    UNDERLINE = '\033[4m'
    RESET = '\033[0m'


print(style.BLUE)
print("**********************************************************")
print("**********************************************************")
print("*****                                               ******")
print("*****  THANK YOU FOR USING WHATSAPP BULK MESSENGER  ******")
print("*****      This tool was built by Anirudh Bagri     ******")
print("*****           www.github.com/anirudhbagri         ******")
print("*****                                               ******")
print("**********************************************************")
print("**********************************************************")
print(style.RESET)

f = open("message.txt", "r")
message = f.read()
f.close()

print(style.YELLOW + '\nThis is your message-')
print(style.GREEN + message)
print("\n" + style.RESET)
message = quote(message)

numbers = []
f = open("numbers.txt", "r")
for line in f.read().splitlines():
    if line.strip() != "":
        numbers.append(line.strip())
f.close()
total_number = len(numbers)
print(style.RED + 'We found ' + str(total_number) +
      ' numbers in the file' + style.RESET)
delay = 30

driver = webdriver.Chrome(ChromeDriverManager().install(), options=options)
print('Once your browser opens up sign in to web whatsapp')
driver.get('https://web.whatsapp.com')
input(style.MAGENTA + "AFTER logging into Whatsapp Web is complete and your chats are visible, press ENTER..." + style.RESET)
for idx, number in enumerate(numbers):
    number = number.strip()
    if number == "":
        continue
    print(style.YELLOW + '{}/{} => Sending message to {}.'.format((idx+1),
                                                                  total_number, number) + style.RESET)
    try:
        url = 'https://web.whatsapp.com/send?phone=' + number + '&text=' + message
        sent = False
        for i in range(3):
            if not sent:
                driver.get(url)
                sleep(15)
                sent = True
                pyautogui.press('enter')
                # try:
                #     click_btn = WebDriverWait(driver, delay).until(
                #         EC.element_to_be_clickable((By.XPATH, "//button[@class='_4sWnG']")))
                # except Exception as e:
                #     print(
                #         style.RED + f"\nFailed to send message to: {number}, retry ({i+1}/3)")
                #     print(
                #         "Make sure your phone and computer is connected to the internet.")
                #     print("If there is an alert, please dismiss it." + style.RESET)
                # else:
                #     sleep(1)
                #     click_btn.click()
                #     sent = True
                #     sleep(3)
                #     print(style.GREEN + 'Message sent to: ' +
                #           number + style.RESET)
    except Exception as e:
        print(style.RED + 'Failed to send message to ' +
              number + str(e) + style.RESET)
driver.close()
`

**Comment by mrityunjayshukla411** [comment]
Install pyautogui
and change the automator.py code to:-

```
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager
from time import sleep
from urllib.parse import quote
import os
import pyautogui

options = Options()
options.add_experimental_option("excludeSwitches", ["enable-logging"])
options.add_argument("--profile-directory=Default")
# options.add_argument("--user-data-dir=/var/tmp/chrome_user_data")

os.system("")
os.environ["WDM_LOG_LEVEL"] = "0"


class style():
    BLACK = '\033[30m'
    RED = '\033[31m'
    GREEN = '\033[32m'
    YELLOW = '\033[33m'
    BLUE = '\033[34m'
    MAGENTA = '\033[35m'
    CYAN = '\033[36m'
    WHITE = '\033[37m'
    UNDERLINE = '\033[4m'
    RESET = '\033[0m'


print(style.BLUE)
print("**********************************************************")
print("**********************************************************")
print("*****                                               ******")
print("*****  THANK YOU FOR USING WHATSAPP BULK MESSENGER  ******")
print("*****      This tool was built by Anirudh Bagri     ******")
print("*****           www.github.com/anirudhbagri         ******")
print("*****                                               ******")
print("**********************************************************")
print("**********************************************************")
print(style.RESET)

f = open("message.txt", "r")
message = f.read()
f.close()

print(style.YELLOW + '\nThis is your message-')
print(style.GREEN + message)
print("\n" + style.RESET)
message = quote(message)

numbers = []
f = open("numbers.txt", "r")
for line in f.read().splitlines():
    if line.strip() != "":
        numbers.append(line.strip())
f.close()
total_number = len(numbers)
print(style.RED + 'We found ' + str(total_number) +
      ' numbers in the file' + style.RESET)
delay = 30

driver = webdriver.Chrome(ChromeDriverManager().install(), options=options)
print('Once your browser opens up sign in to web whatsapp')
driver.get('https://web.whatsapp.com')
input(style.MAGENTA + "AFTER logging into Whatsapp Web is complete and your chats are visible, press ENTER..." + style.RESET)
for idx, number in enumerate(numbers):
    number = number.strip()
    if number == "":
        continue
    print(style.YELLOW + '{}/{} => Sending message to {}.'.format((idx+1),
                                                                  total_number, number) + style.RESET)
    try:
        url = 'https://web.whatsapp.com/send?phone=' + number + '&text=' + message
        sent = False
        for i in range(3):
            if not sent:
                driver.get(url)
                sleep(15)
                sent = True
                pyautogui.press('enter')
                # try:
                #     click_btn = WebDriverWait(driver, delay).until(
                #         EC.element_to_be_clickable((By.XPATH, "//button[@class='_4sWnG']")))
                # except Exception as e:
                #     print(
                #         style.RED + f"\nFailed to send message to: {number}, retry ({i+1}/3)")
                #     print(
                #         "Make sure your phone and computer is connected to the internet.")
                #     print("If there is an alert, please dismiss it." + style.RESET)
                # else:
                #     sleep(1)
                #     click_btn.click()
                #     sent = True
                #     sleep(3)
                #     print(style.GREEN + 'Message sent to: ' +
                #           number + style.RESET)
    except Exception as e:
        print(style.RED + 'Failed to send message to ' +
              number + str(e) + style.RESET)
driver.close()

```

**Comment by akhilrajs** [comment]
bro i have made an update to your code . check it out here : https://github.com/akhilrajs/Whatsapp-messaging-BOT

**Comment by kmnewazbillah** [comment]
Please fixed the sending button not working function

**Comment by ahmadkeren** [comment]
button and list message only work at official api

**Comment by anirudhbagri** [comment]
This should be fixed now.
